File: ArchiRedux/ArchiRedux/ClientApp/src/model/product.ts
export class Product {
    public id: string;
    
    public name: string;
    
    public price: number;

    public isRemoved: boolean;

    constructor(id: string, name: string, price: number, isRemoved: boolean) {
        this.id = id;
        this.name = name;
        this.price = price;
        this.isRemoved = isRemoved;
    }
}

File: ArchiRedux/ArchiRedux/Domain/Model/Product.cs
using System;

namespace ArchiRedux.Domain.Model
{
    public class Product
    {
        public Guid Id { get; set; }
        public String Name { get; set; }
        public Int32 Price { get; set; }
        public Boolean IsRemoved { get; set; }

        public Product() { }

        public Product(Guid id, string name, int price, bool isRemoved)
        {
            Id = id;
            Name = name;
            Price = price;
            IsRemoved = isRemoved;
        }
    }
}